drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -6009,7 +5999,7 @@ Error RenderingDeviceVulkan::buffer_update(RID p_buffer, uint32_t p_offset, uint
 	// no barrier should be needed here
 	// _buffer_memory_barrier(buffer->buffer, p_offset, p_size, dst_stage_mask, VK_PIPELINE_STAGE_TRANSFER_BIT, dst_access, VK_ACCESS_TRANSFER_WRITE_BIT, true);
 
-	Error err = _buffer_update(buffer, p_offset, (uint8_t *)p_data, p_size, p_post_barrier);
+	Error err = _buffer_update(buffer, p_offset, (uint8_t *)p_data, p_size, p_post_barrier != RD::BARRIER_MASK_NO_BARRIER);
```

Comment:
This change is my biggest point of doubt.

Before the change, `p_post_barrier` would always booleanize to `true` since even `BARRIER_MASK_NO_BARRIER` is non-zero.

Therefore, I thought that the changed code fits the original intent better.

Reply:
This makes sense, but I dont remember why it was used. I will have to check. For buffers it probably makes no difference, though?

Reply:
By the way, wouldn't it be better if `BARRIER_MASK_NO_BARRIER` was `0``?

Reply:
Well, I've just pushed with such change in an additional commit.